action.yml

```diff
@@ -98,36 +98,35 @@ runs:
 
     - name: Build dependencies.log (and commit)
       run: |
-        pip3 install pip-tools
-
-        # do main dependencies.log in subproc
-        echo
-        echo "dependencies.log"
-        git mv requirements.txt dependencies.log || true
-        echo "pip-compile..."
-        pip-compile --upgrade --output-file="dependencies.log" &
-
-        # get all extras
-        EXTRAS=$(python3 ${{ github.action_path }}/list_extras.py setup.cfg)
-
-        # generate dependencies-*.log for each extras_require (each in a subproc)
-        for extra in $EXTRAS; do
-          echo
-          extra_dep_log="dependencies-${extra}.log"
-          echo $extra_dep_log
-          git mv "requirements-${extra}.txt" $extra_dep_log || true
-          echo "pip-compile..."
-          pip-compile --upgrade --extra $extra --output-file="$extra_dep_log" &
-        done
-        echo
+        # append permissive line to .gitignore since *.log is commonly present
+        line='!dependencies*.log'
+        if [[ ! $(grep -F "$line" .gitignore) ]]; then
+          echo "" >> .gitignore
+          echo "" >> .gitignore
+          echo "# wipac-dev-py-setup-action" >> .gitignore
+          echo "$line" >> .gitignore
+          git add .gitignore
+          git commit -m "<bot> update .gitignore" || true  # fails if no change
+          tail .gitignore
+        fi
 
-        # wait for all subprocs
-        wait -n # main dependencies.log
-        for extra in $EXTRAS; do
-          wait -n
-        done
+        # remove any old ones, then regenerate only what's needed
+        rm dependencies*.log
+
+        # Build
+        if [ -f ./Dockerfile ]; then
+          # from docker...
+          for f in ./Dockerfile*; do
+              echo $f
+              ${{ github.action_path }}/build-dependencies-dockerfile-logs.sh $f
+          done
+        else
+          # from setup.cfg...
+          export GITHUB_ACTION_PATH=${{ github.action_path }}
+          ${{ github.action_path }}/build-dependencies-logs.sh
+        fi
 
-        # commit each changed file individually
+        # Commit each changed file individually
```

Comment:
I'm debating if it's better to do this all in one commit, since it was a single instance of dependency scanning.  Also, that's a lot of commits when you have a lot of extras and Dockerfiles.

Reply:
That's a good point, it wouldn't hurt to not do these separately